### DISCUSSION TITLE
Workflow Support and Hidden Fields

### DISCUSSION THREAD

**Discussion by ninjz** [payloadcms/payload#2930] [Q&A]
I'm evaluating CMS solutions and PayloadCMS and Sanity are my top picks at the moment. Something I feel is a requirement for us is something like this Workflow plugin that Sanity has:
https://www.npmjs.com/package/sanity-plugin-workflow

Sanity also has support for Hidden fields: https://www.sanity.io/docs/conditional-fields#82aaf3c8dbb2

I am coming from using Airtable as a temporary CMS which allows us to have a workflow that contains hidden fields for different editors of content and different status lanes as we have a complex creation + review process. For example we have fields that are reference comments and videos which tell an animator how to create an animation which then an animator uses this to create the animation, which then later goes on to another review/feedback cycle. The final content returns the public fields, and these reference comments and videos are hidden.

Possibly, this process can be managed in Airtable then the completed content can be inputted into the CMS, but I would much rather we can have this entire workflow integrated into the CMS directly. I would prefer Payload over Sanity if this is supported.

Is there something similar with Payload or is this something that can easily be extended to support?

**Comment by jmikrut** [comment]
Hey @ninjz — thanks for stopping by! Here are a few answers for you.

#### Workflows

We do offer a Publication Workflows plugin which is similar to what you've linked to from Sanity. It's an Enterprise feature, but if you are not looking for an enterprise license, you can certainly build something like this yourself directly within Payload. If you're interested in hearing more about Payload Enterprise, we'd be happy to chat! Just fill out the form on [this page](https://payloadcms.com/for-enterprise) and we can follow up with you immediately. 

From your description, I suspect that your use case is quite specific, and you might benefit from the flexibility that Payload offers here for sure. Our existing Enterprise plugin could get you what you need, but we also would love to extend it so that it literally delivers exactly what you need.

#### Hidden fields

Interesting, I never knew that Sanity had this (admittedly, I don't look too closely at them). We do too, of course. It's built in directly and it's called `condition` in Payload. [Here are the docs](https://payloadcms.com/docs/fields/overview#conditional-logic).

I would be pumped to talk more about your specific workflow needs, so please do feel free to fill out that form I linked above if you're interested in collaborating on this.

Thanks again for stopping by!

**Reply by ninjz** [reply]
Hey @jmikrut, thank you for your response. I would say we're likely not looking for an enterprise license at this moment (we're a small early stage startup and that sounds a bit overkill for our needs).

Would there be any possibility for this Publication Workflow plugin to be released as a purchasable add-on for non-enterprise plans in the near future? Otherwise I think the choice is quite clear for us.

**Reply by DennisSnijder** [reply]
Hi @ninjz Not sure if this helps you on your decision, but I'm currently working on such a workflow plugin for Payload. I hope to release a first version very soon. I'm posting progress on the Payload discord server if you're interested. 

Don't think it will be anywhere near the feature completeness of Sanity's workflow plugin upon first release, but it will be a start 😄.
I'll make sure to link the plugin when I released a first version of it! 

![Screenshot 2023-06-30 at 11 14 36](https://github.com/payloadcms/payload/assets/6953846/60316aa4-266f-4ed8-8b18-90b359ef60e3)

**Reply by DennisSnijder** [reply]
@ninjz as promised 🙂

https://github.com/DennisSnijder/payload-workflow

**Reply by ninjz** [reply]
This looks great @DennisSnijder. Thank you for this! I will give this a try and see if it fits our needs.

**Reply by DennisSnijder** [reply]
@ninjz Nice! if you have any recommendation or want a specific feature, feel free to open an issue on the repository. 😄
